src/decorator/typechecker/IsEnum.ts

```diff
@@ -11,6 +11,15 @@ export function isEnum(value: unknown, entity: any): boolean {
   return enumValues.indexOf(value) >= 0;
 }
 
+/**
+ * Returns an array of all the possible values for a given enum
+ */
+export function validEnumValues(entity: any): any[] {
+  return Object.entries(entity)
+    .filter(entry => !parseInt(entry[0]))
```

Comment:
Implicitly defined enums actually start with `0` and due to type coercion this check is not reliable.

If you define 
```typescript
  enum MyEnum {
    First,
    Second,
  }
```
one of the entries is `['0', 'First']`, and `!parseInt('0') === !0 === true`
and the return value for `MyEnum` will be `['First', '0', 1']` instead of the expected `['0', '1']`.

It would be more reliable to check whether the key is not a number after all, like:

```suggestion
    .filter(entry => isNaN(parseInt(entry[0])))
```